x-pack/plugins/security_solution/public/common/components/event_details/endpoint_response_actions_tab.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiNotificationBadge, EuiSpacer } from '@elastic/eui';
+import { ActionsLogTable } from '../../../management/components/endpoint_response_actions_list/components/actions_log_table';
+import { useGetEndpointActionList } from '../../../management/hooks';
+import type { AlertRawEventData, ExpandedEventFieldsObject } from './types';
+import { EventsViewType } from './event_details';
+import * as i18n from './translations';
+
+import { expandDottedObject } from '../../../../common/utils/expand_dotted';
+import { useIsExperimentalFeatureEnabled } from '../../hooks/use_experimental_features';
+import { RESPONSE_ACTION_TYPES } from '../../../../common/detection_engine/rule_response_actions/schemas/response_actions';
+
+const TabContentWrapper = styled.div`
+  height: 100%;
+  position: relative;
+`;
+
+export const useEndpointResponseActionsTab = ({
+  rawEventData,
+}: {
+  rawEventData?: AlertRawEventData;
+}) => {
+  const responseActionsEnabled = useIsExperimentalFeatureEnabled('endpointResponseActionsEnabled');
+
+  const {
+    data: actionList,
+    isFetching,
+    isFetched,
+  } = useGetEndpointActionList(
+    {
+      alertIds: [rawEventData?._id ?? ''],
+      withRuleActions: true,
+    },
+    { retry: false, enabled: rawEventData && responseActionsEnabled }
+  );
+
+  const totalItemCount = useMemo(() => actionList?.total ?? 0, [actionList]);
+
+  if (!rawEventData || !responseActionsEnabled) {
+    return;
+  }
+
+  const expandedEventFieldsObject = expandDottedObject(
```

Comment:
can we move it higher and pass just filtered response actions to this component?